src/request_body_processor/multipart.cc

```diff
@@ -61,27 +66,35 @@ MultipartPartTmpFile::~MultipartPartTmpFile() {
 
 void MultipartPartTmpFile::Open() {
     struct tm timeinfo;
-    char tstr[300];
     time_t tt = time(NULL);
 
     localtime_r(&tt, &timeinfo);
 
-    memset(tstr, '\0', 300);
+    char tstr[300] {};
```

Comment:
SonarCloud [complains](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=3132&id=owasp-modsecurity_ModSecurity&open=AY8WiZ1ySIO09gNZIKyb) for this solution, we have to take a look it later. (I see the `strftime` later). Actually you don't need to touch this.

Reply:
Yes, you could just do away with the initialization of the buffer altogether as it's being initialized in the next line, so zero-initialization is not needed here.

The buffer size is actually a bit excessive, as the format string needs 2 chars (for `/` & `-`), 8 for the date (`%Y%m%d`), 6 for the time (`%H%M%S`) and 1 more for the null terminator, so a size of 17 should be enough.

```cpp
char tstr[17];
strftime(tstr, sizeof(tstr)/sizeof(tstr[0]), "/%Y%m%d-%H%M%S", &timeinfo);
```

I commented on the SonarCloud reported issue before [here](https://github.com/owasp-modsecurity/ModSecurity/pull/3132#issuecomment-2079836460), which I think is a false positive.

Reply:
I mean SonarCloud complains it before we use a C-style string here, and it suggest to use C++'s `std::string`. But the `strftime()` expects a `char *` so actually I don't know how could it be resolved. As you wrote here, we should mark it as false positive.